be/src/olap/task/index_builder.cpp

```diff
@@ -198,6 +198,11 @@ Status IndexBuilder::handle_single_rowset(RowsetMetaSharedPtr output_rowset_meta
                 }
             }
 
+            if (return_columns.empty()) {
+                // no columns to read
+                break;
```

Comment:
Do you mean continue? break will skip remaining segments that may have return_columns.

Reply:
return_columns is build by _alter_inverted_indexes, so the behavior of all segments is consistent, so here can be directly break